src/aCorr.cu

```diff
@@ -0,0 +1,358 @@
+
+/*
+ * Copyright (c) 2018-2020, The Bifrost Authors. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ * * Neither the name of The Bifrost Authors nor the names of its
+ *   contributors may be used to endorse or promote products derived
+ *   from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* 
+
+ 
+
+Implements the grid-Correlation onto a GPU using CUDA. 
+
+*/
+#include <iostream>
+#include "bifrost/aCorr.h"
+#include "assert.hpp"
+#include "trace.hpp"
+#include "utils.hpp"
+#include "cuda.hpp"
+#include "cuda/stream.hpp"
+//#include <complex.h>
+#include "Complex.hpp"
+
+#include <thrust/device_vector.h>
+
+#define tile 256 // Number of threads per thread-block
+
+/********************/
+/* CUDA ERROR CHECK */
+/********************/
```

Comment:
Why are these checks happening here but not in your other modules? Should they happen everywhere, or is this one special for some reason?

Reply:
They seem similar to the `BF_CHECK_CUDA_EXCEPTION` and `BF_CHECK_CUDA` macros in `cuda.hpp`.